redwood-lazylayout-widget/src/commonTest/kotlin/app/cash/redwood/lazylayout/widget/LazyListUpdateProcessorTest.kt

```diff
@@ -566,6 +566,89 @@ class LazyListUpdateProcessorTest {
     assertThat(processor.toString()).isEqualTo("[2...] . . . . [...13]")
   }
 
+  /**
+   * We had a bug where we attempted to move the split between `itemsBefore` and `itemsAfter` beyond
+   * the end of `itemsAfter`. This bug would occurs when the following held:
+   *
+   *  * The items before count is shrinking
+   *  * The insert size is greater than all unloaded items (itemsBefore + itemsAfter)
+   *
+   * https://github.com/cashapp/redwood/issues/2172
+   */
+  @Test
+  fun shiftItemsFromAfterToBeforeWhenThereAreNotEnoughToShift() {
+    processor.itemsBefore(3)
+    processor.itemsAfter(3)
+    processor.onEndChanges()
+    processor.scrollTo(0, 6)
+    assertThat(processor.toString()).isEqualTo(". . . . . .")
+
+    processor.itemsBefore(2)
+    processor.items.insert(0, "C")
+    processor.items.insert(1, "D")
+    processor.items.insert(2, "E")
+    processor.items.insert(3, "F")
+    processor.items.insert(4, "G")
+    processor.items.insert(5, "H")
+    processor.items.insert(6, "I")
+    processor.onEndChanges()
+    processor.scrollTo(0, 12)
+    assertThat(processor.toString()).isEqualTo(". . C D E Fv2 G H I . . .")
```

Comment:
The `v2` after `F` here means there was an update-in-place, rather than a structural change. I’d like to do a follow-up to make these all say `v2`, which is how we reduce the glitchy animations in update processing.